src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,16 +28,17 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        synchronized (arguments) {
```

Comment:
@ChristianSchwarz that makes sence. Sorry for taking me so much time to understand that. I introduced locks to prevent this, can you please have a look?